### PR TITLE
Add num_nodes argument

### PR DESCRIPTION
**Pull request by janEbert**
Make code run correctly in the multi-node setting.

This mainly introduces the `num_nodes` argument ~~and fixes all
occurrences of `TrainArgs` method calls that were previously only
supplied the number of devices instead of the world size (which we now
access using `fabric.world_size`)~~.

~~Because we no longer need to pass the `devices` integer around, this
argument is removed from all signatures.~~

~~I also took the liberty to rename the `devices` argument of the methods
of `TrainArgs` to `world_size` for future clarity.~~

### CODE REVIEW

**Review by rasbt — APPROVED** [review]
This PR looks awesome. Thanks for doing this! 
It looks all good to me, but I am CCing @awaelchli for a second look because he is the multi-GPU, multi-node, and Fabric expert. Adrian, if you could take a look, that'd be great!

**Comment by janEbert** [comment]
Thanks for updating all the configs!

**Review by awaelchli — COMMENTED** [review]
Hi @janEbert 

I'm ok with adding the `num_nodes` as an argument that the user can manually set. So far, we've done multi-node training with LitGPT on Lightning AI where this is not required at all to be set. It gets inferred automatically. 
However, I would definitely not change the definitions of batch size and gradient accumulation without completing the discussion. The name is maybe not appropriate, but the choice of how to set it so that it's practical and easy to use should be taken into consideration. You can take a look at my reasoning and suggestion there: https://github.com/Lightning-AI/litgpt/issues/1474

File: litgpt/args.py [hunk 1]
```diff
@@ -50,24 +50,24 @@ def __post_init__(self) -> None:
...
-        gradient_accumulation_iters = self.batch_size(devices) // self.micro_batch_size
+        gradient_accumulation_iters = self.batch_size(world_size) // self.micro_batch_size
         assert gradient_accumulation_iters > 0
         return gradient_accumulation_iters
 
-    def batch_size(self, devices: int) -> int:
+    def batch_size(self, world_size: int) -> int:
         """Number of samples between optimizer steps per data-parallel rank"""
-        batch_size = self.global_batch_size // devices
+        batch_size = self.global_batch_size // world_size
```
[the remark this answers is not in the document]
Comment (janEbert) [reply litgpt/args.py:61]: Thanks for pointing to the discussion regarding keeping the previous behavior. I'll revert all changes related to `TrainArgs` and their method calls, then?

[the remark this answers is not in the document]
Comment (awaelchli) [reply litgpt/args.py:61]: Yes, reverting the changes to the batch size related methods in args.py and calls in the script would be great, but it might be a lot of work to do manually, sorry. 

I think adding num_nodes is great. Setting the env var is not meant to be done by the user but by the launcher, so I would not do that.

[the remark this answers is not in the document]
Comment (janEbert) [reply litgpt/args.py:61]: Sounds good, thanks for the feedback!

[the remark this answers is not in the document]
Comment (janEbert) [reply litgpt/args.py]: I removed it all, PTAL.

**Review by awaelchli — APPROVED** [review]

File: litgpt/args.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (awaelchli) [reply litgpt/args.py]: Thanks for making the adjustments 🎉

**Comment by rasbt** [comment]
Looks great, thanks everyone!

**Review by rasbt — APPROVED** [review]
